src/Middleware/DetectLocaleMiddleware.php

```diff
@@ -186,8 +186,12 @@ protected function getPersistKey()
     protected function getParamLocale(HTTPRequest $request)
     {
         $queryParam = FluentDirectorExtension::config()->get('query_param');
-        $locale = (string)$request->param($queryParam) ?: $request->requestVar($queryParam);
-        return $locale;
+        if (FluentDirectorExtension::config()->get('detect_locale')
```

Comment:
detect_locale is meant to be a flag for "make the guess guess if we don't have an explicit locale". In this case a query param is explicit so we don't need to treat it as though we would an ambiguous check.